app/main.py

```diff
@@ -1,34 +1,61 @@
+from typing import Optional
+
+
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self, row: int, column: int, is_alive: bool = True) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(self, start: tuple,
+                 end: tuple,
+                 is_drowned: bool = False) -> None:
+
+        self.decks = []
+        self.start = start
+        self.end = end
+        self.is_drowned = is_drowned
+        self.create_deck()
+
+    def create_deck(self) -> None:
+        start_point = [self.start[0], self.start[1]]
+        self.decks.append(Deck(start_point[0], start_point[1]))
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+        while start_point[0] != self.end[0]:
+            start_point[0] += 1
+            self.decks.append(Deck(start_point[0], start_point[1]))
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+        while start_point[1] != self.end[1]:
+            start_point[1] += 1
+            self.decks.append(Deck(start_point[0], start_point[1]))
+
+    def get_deck(self, row: int, column: int) -> Optional[Deck]:
+        for deck in self.decks:
+            if deck.row == row and deck.column == column:
+                return deck
+
+    def fire(self, row: int, column: int) -> None:
+        self.get_deck(row, column).is_alive = False
+        self.change_is_drowned()
+
+    def change_is_drowned(self) -> None:
+        self.is_drowned = not any(deck.is_alive for deck in self.decks)
 
 
 class Battleship:
-    def __init__(self, ships):
-        # Create a dict `self.field`.
-        # Its keys are tuples - the coordinates of the non-empty cells,
-        # A value for each cell is a reference to the ship
-        # which is located in it
-        pass
-
-    def fire(self, location: tuple):
-        # This function should check whether the location
-        # is a key in the `self.field`
-        # If it is, then it should check if this cell is the last alive
-        # in the ship or not.
-        pass
+    def __init__(self, ships: list[tuple]) -> None:
+        self.field = {}
+        for ship in ships:
+            boat = Ship(ship[0], ship[1])
+            for deck in boat.decks:
+                self.field[(deck.row, deck.column)] = boat
+
+    def fire(self, location: tuple) -> str:
+        if location not in self.field.keys():
```

Comment:
It's not necessary to use keys() here, because `in self.field` is using keys only by default.